static/js/settings.js

```diff
@@ -238,20 +238,23 @@ var SettingsScreen = {
     }).then(function() {
       updateNow.textContent = 'In Progress'
       let isDown = false;
-      let interval = setInterval(function() {
+      function checkStatus() {
         API.getUpdateStatus().then(function() {
           if (isDown) {
-            clearInterval(interval);
             SettingsScreen.showUpdateSettings();
             updateNow.textContent = 'Update Now';
+          } else {
+            setTimeout(checkStatus, 5000);
           }
         }).catch(function() {
           if (!isDown) {
             updateNow.textContent = 'Restarting';
             isDown = true;
           }
+          setTimeout(checkStatus, 5000);
         });
-      }, 1000);
+      }
+      checkStatus();
     }).catch(function() {
       updateNow.textContent = 'Error';
     });
```

Comment:
So how frequently does this wind up checking github? I'd guess that we should probably only check a few times (maybe even just once) per day.

We should probably also check if the user requests a check.

Reply:
This ends up checking every time the page is loaded, which is how I got rate-limited by GitHub. I definitely want to add some form of caching before landing this